src/uct/ib/rc/verbs/rc_verbs_ep.c

```diff
@@ -449,6 +450,81 @@ ucs_status_t uct_rc_verbs_ep_fence(uct_ep_h tl_ep, unsigned flags)
     return uct_rc_ep_fence(tl_ep, &ep->fi, 1);
 }
 
+static ucs_status_t uct_rc_verbs_ep_check_internal(uct_ep_h tl_ep)
+{
+    uct_rc_verbs_ep_t *ep       = ucs_derived_of(tl_ep, uct_rc_verbs_ep_t);
+    uct_rc_verbs_iface_t *iface = ucs_derived_of(tl_ep->iface,
+                                                 uct_rc_verbs_iface_t);
+
+    /* in case if no TX resources are available then there is at least
+     * one signaled operation which provides actual peer status, in this case
+     * just return without any actions */
+    UCT_RC_CHECK_TXQP_RET(&iface->super, &ep->super, UCS_OK);
+    UCT_RC_CHECK_CQE_RET(&iface->super, &ep->super, UCS_ERR_NO_RESOURCE);
+
+    return uct_rc_verbs_ep_post_flush(ep, 0);
```

Comment:
`uct_rc_verbs_ep_post_flush` will check resources again, is it possible to avoid that?